File: 01/one.hs
-- | Parse a frequency shift such as "+12" or "-15". 'read' chokes on a
-- leading plus sign, so strip it off first.
parseShift :: String -> Int
parseShift ('+' : digits) = read digits
parseShift shift = read shift

parseShifts :: String -> [Int]
parseShifts = map parseShift . filter (not . null) . lines

-- | Sum every frequency shift to get the resulting frequency.
calc :: [Int] -> Int
calc = sum

main :: IO ()
main = do
  input <- readFile "input.txt"
  print (calc (parseShifts input))

File: 01/one.rs
use std::fs;

fn calc(input: &str) -> i64 {
    input
        .lines()
        .filter(|line| !line.trim().is_empty())
        .map(|line| line.trim().parse::<i64>().unwrap())
        .sum()
}

fn main() {
    let input = fs::read_to_string("input.txt").expect("failed to read input.txt");
    println!("{}", calc(&input));
}
